### PR TITLE
[DNM]*: init loader tool project.

### PR DESCRIPTION
**Pull request by qiuyesuifeng**

### CODE REVIEW

File: loader/loader.go [hunk 1]
```diff
@@ -0,0 +1,520 @@
...
+	// mydumper file names format
+	// db -> {db}-schema-create.sql
+	// table -> {db}.{table}-schema.sql
+	// sql -> {db}.{table}.sql
+
+	usedFiles := make(map[string]struct{})
+
+	// scan db schema files
+	for file, _ := range files {
+		idx := strings.Index(file, "-schema-create.sql")
```
Comment (zhangjinpeng1987) [inline loader/loader.go]: end with

File: loader/loader.go [hunk 2]
```diff
@@ -0,0 +1,520 @@
...
+	}
+
+	// scan db table schema files
+	for file, _ := range files {
+		_, ok := usedFiles[file]
+		if ok {
+			continue
+		}
+
+		idx := strings.Index(file, "-schema.sql")
```
Comment (zhangjinpeng1987) [inline loader/loader.go]: end with

File: loader/loader.go [hunk 3]
```diff
@@ -0,0 +1,520 @@
...
+		if idx > 0 {
+			name := file[:idx]
+			fields := strings.Split(name, ".")
+			if len(fields) < 2 {
```
Comment (zhangjinpeng1987) [inline loader/loader.go]: len(fields) != 2

File: loader/loader.go [hunk 4]
```diff
@@ -0,0 +1,520 @@
...
+	}
+
+	// scan db table schema files
+	for file, _ := range files {
+		_, ok := usedFiles[file]
+		if ok {
+			continue
+		}
+
+		idx := strings.Index(file, ".sql")
```
Comment (zhangjinpeng1987) [inline loader/loader.go]: how about `-schema-view.sql` `-schema-triggers.sql` `-schema-post.sql`?

File: loader/loader.go [hunk 5]
```diff
@@ -0,0 +1,520 @@
...
+			}
+
+			time.Sleep(waitTime)
+		}
+	}
+}
+
+func (l *Loader) run() error {
+	var err error
+	l.dbs, err = createDBs(l.cfg.DB, l.cfg.Worker)
```
Comment (zhangjinpeng1987) [inline loader/loader.go]: s/l.dbs/l.conns/

File: loader/loader.go [hunk 6]
```diff
@@ -0,0 +1,520 @@
...
+					sqls = append(sqls, fmt.Sprintf("use %s;", schema))
+				}
+
+				sqls = append(sqls, query)
+				err = executeSQL(db, sqls, false)
+				if err != nil {
+					return errors.Trace(err)
+				}
+
+				data = data[0:0]
```
Comment (zhangjinpeng1987) [inline loader/loader.go]: ?

File: loader/loader.go (hunk 4 above)
Comment (qiuyesuifeng) [inline loader/loader.go]: now tidb has not supported view and trigger, what is `-schema-post`?

File: loader/loader.go [hunk 7]
```diff
@@ -0,0 +1,607 @@
...
+				for _ = range sqls {
+					l.jobWg.Done()
+				}
+
+				idx = 0
+				sqls = sqls[0:0]
+				lastSyncTime = now
+			}
+
+			time.Sleep(waitTime)
```
Comment (tiancaiamao) [inline loader/loader.go]: Personally, I prefer a timeout ticker.C to handle think kind of logic

File: loader/loader.go [hunk 8]
```diff
@@ -0,0 +1,607 @@
...
+	}
+
+	log.Infof("[loader][save meta]%s", l.meta)
+
+	log.Infof("[loader][run table data sql]%s[end]", file)
+	return nil
+}
+
+func (l *Loader) run() error {
+	var err error
```
Comment (tiancaiamao) [inline loader/loader.go]: there are many log.Fatal in this function, maybe defer and save status make it more graceful ?

**Pull request closed without merging**